driver/internal/protocol/authentication_jwt.go

```diff
@@ -0,0 +1,226 @@
+
+	return size
+}
+
+func (r *authInitReqJWT) encode(enc *encoding.Encoder) error {
+	enc.Int16(int16(1 + 1*2)) // empty username + one method with two fields (method name + token)
+
+	// empty user name
+	if err := authShortCESU8String.encode(enc, ""); err != nil {
+		return err
+	}
+
+	return r.method.encode(enc)
+}
+
+func (r *authInitReqJWT) decode(dec *encoding.Decoder, ph *partHeader) error {
+	panic("not implemented")
+}
+
+type authInitRepJWT struct {
+	username string
+}
+
+func (r *authInitRepJWT) String() string {
+	return fmt.Sprintf("method %s user %v", mnJWT, r.username)
+}
+
+func (r *authInitRepJWT) size() int {
+	panic("not implemented")
+}
+
+func (r *authInitRepJWT) encode(enc *encoding.Encoder) error {
+	panic("not implemented")
+}
+
+func (r *authInitRepJWT) decode(dec *encoding.Decoder, ph *partHeader) error {
+	numPrm := int(dec.Int16())
+	if numPrm != 2 {
+		return fmt.Errorf("invalid number of parameters %d - expected %d", numPrm, 2)
+	}
+
+	method := string(authShortBytes.decode(dec))
+	if err := checkMethodJWT(method); err != nil {
+		return err
+	}
+
+	var err error
+	r.username, err = authShortCESU8String.decode(dec)
+
+	return err
+}
+
+type authFinalReqJWT struct {
+	username string
+}
+
+func (r *authFinalReqJWT) String() string {
+	return fmt.Sprintf("username %s method %s emptyparam", r.username, mnJWT)
+}
+
+func (r *authFinalReqJWT) size() int {
+	size := int16Size // no of parameters
+
+	// username
+	size++ // len byte username
+	size += cesu8.StringSize(r.username)
+
+	// method name
+	size++ // len byte method name
+	size += len(mnJWT)
+
+	size++ // len byte empty parameter
+
+	return size
+}
+
+func (r *authFinalReqJWT) decode(dec *encoding.Decoder, ph *partHeader) error {
+	panic("not implemented")
+}
+
+func (r *authFinalReqJWT) encode(enc *encoding.Encoder) error {
+	enc.Int16(3) // username, method, empty parameter
+
+	if err := authShortCESU8String.encode(enc, r.username); err != nil {
+		return err
+	}
+
+	if err := authShortBytes.encode(enc, []byte(mnJWT)); err != nil {
+		return err
+	}
+
+	if err := authShortBytes.encode(enc, []byte{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type authFinalRepJWT struct {
+	cookie []byte
+}
+
+func (r *authFinalRepJWT) String() string {
+	return fmt.Sprintf("method %s cookie %v", mnJWT, r.cookie)
+}
+
+func (r *authFinalRepJWT) size() int {
+	panic("not implemented")
+}
+
+func (r *authFinalRepJWT) encode(enc *encoding.Encoder) error {
+	panic("not implemented")
+}
+
+func (r *authFinalRepJWT) decode(dec *encoding.Decoder, ph *partHeader) error {
+	numPrm := int(dec.Int16())
+	if numPrm != 2 {
+		return fmt.Errorf("invalid number of parameters %d - expected %d", numPrm, 2)
+	}
+
+	method := string(authShortBytes.decode(dec))
+	if err := checkMethodJWT(method); err != nil {
+		return err
+	}
+
+	cookieSize, null := decodeVarBytesSize(dec)
+	if !null {
+		r.cookie = make([]byte, cookieSize) // should we do something with the cookie?
```

Comment:
the server returns a cookie. Not sure if we need to do something with it.

Reply:
If someone later implements the SessionCookie authentication method, the username+cookie could be used reconnect.